lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -2349,28 +2436,132 @@ void HdVP2Material::_UpdateShaderInstance(const HdMaterialNetwork& mat)
+        if (fallbackIt != node.parameters.end()) {
+            value = fallbackIt->second;
+        }
+
+        MHWRender::MTexture* texture = _LoadTexture(path, isSRGB, uvScaleOffset, value);
+
+        HdVP2TextureInfo& info = _textureMap[path];
+        info._texture.reset(texture);
+        info._isColorSpaceSRGB = isSRGB;
+        if (uvScaleOffset.length() > 0) {
+            TF_VERIFY(uvScaleOffset.length() == 4);
+            info._stScale.Set(
+                uvScaleOffset[0], uvScaleOffset[1]); // The first 2 elements are the scale
+            info._stOffset.Set(
+                uvScaleOffset[2], uvScaleOffset[3]); // The next two elements are the offset
+        }
+
+        return info;
+    }
+
+    VtValue fallbackValue;
+    auto    fallbackIt = node.parameters.find(_tokens->fallback);
+    if (fallbackIt != node.parameters.end()) {
+        fallbackValue = fallbackIt->second;
+    }
+
+    {
+        std::lock_guard<std::mutex> lock(_textureLoadingMutex);
+        _textureLoadingTasks.emplace(
+            path, new TextureLoadingTask(this, sceneDelegate, path, fallbackValue));
+    }
+
+    static HdVP2TextureInfo dummyInfo;
+
+    if (!dummyInfo._texture) {
+        // Create a default texture info with fallback color
+        MHWRender::MRenderer* const       renderer = MHWRender::MRenderer::theRenderer();
+        MHWRender::MTextureManager* const textureMgr
+            = renderer ? renderer->getTextureManager() : nullptr;
+        if (textureMgr) {
+            GfVec4f fallbackColor { 0.8, 0.8, 0.8, 0 };
+            if (fallbackIt != node.parameters.end() && fallbackIt->second.IsHolding<GfVec4f>()) {
+                fallbackColor = fallbackIt->second.UncheckedGet<GfVec4f>();
+            }
+            dummyInfo._texture.reset(
+                _GenerateFallbackTexture(textureMgr, "dummy", VtValue(fallbackColor)));
+        }
+    }
+
+    return dummyInfo;
+}
+
+void HdVP2Material::LoadTextures()
+{
+    std::lock_guard<std::mutex> lock(_textureLoadingMutex);
+    for (const auto& task : _textureLoadingTasks) {
+        task.second->Load();
+    }
+}
+
+void HdVP2Material::ClearPendingTasks()
+{
+    // Inform tasks that have not started or finished that this material object
+    // is no longer valid
+    std::lock_guard<std::mutex> lock(_textureLoadingMutex);
+
+    for (auto& task : _textureLoadingTasks) {
+        if (task.second->Terminate()) {
+            // Delete the pointer: we can only do that outside of the object scope
+            delete task.second;
+        }
+    }
+    // Remove the reference of all the tasks
+    _textureLoadingTasks.clear();
+}
+
+void HdVP2Material::_UpdateLoadedTexture(
+    HdSceneDelegate*     sceneDelegate,
+    const std::string&   path,
+    MHWRender::MTexture* texture,
+    bool                 isColorSpaceSRGB,
+    const MFloatArray&   uvScaleOffset)
+{
+    {
+        std::lock_guard<std::mutex> lock(_textureLoadingMutex);
```

Comment:
I think we also don't need to acquire the _textureLoadingMutex here. This is called from the idle task and nothing will be running in parallel.

Reply:
Removed lock.